GameServer.cpp

```diff
@@ -127,22 +126,29 @@ void GameServer::AllocSocket(SOCKET& hostSocket, unsigned short port)
 	int errorCode = bind(hostSocket, (sockaddr*)&hostAddress, sizeof(hostAddress));
 
 	if (errorCode == SOCKET_ERROR) {
-		throw std::runtime_error("Unable to bind socket at host address of " + 
-			FormatIPAddressWithPort(hostAddress.sin_addr.s_addr, port) +
-			". Windows Sockets Error Code: " + std::to_string(WSAGetLastError()));
+		throw std::runtime_error(FormatSocketError("Unable to bind socket at host address of ", hostAddress));
 	}
 
 	// Set non-blocking mode
 	unsigned long argp = true;
 	errorCode = ioctlsocket(hostSocket, FIONBIO, &argp);
 
 	if (errorCode != 0) {
-		throw std::runtime_error("Unable to set non-blocking mode on socket at host address of  " + 
-			FormatIPAddressWithPort(hostAddress.sin_addr.s_addr, port) + 
-			". Windows Sockets Error Code: " + std::to_string(WSAGetLastError()));
+		throw std::runtime_error(FormatSocketError("Unable to set non-blocking mode on socket at host address of  ", hostAddress));
 	}
 }
 
+std::string GameServer::FormatSocketError(const std::string& message, const sockaddr_in& address)
+{
+	std::string formattedMessage = message + FormatIPAddressWithPort(address.sin_addr.s_addr, address.sin_port) + ". ";

```

Comment:
I was kind of thinking the `FormatIPAddressWithPort` method could have taken a single `const sockaddr_in&` parameter, and formatted both the address and port from that. Wouldn't need the second parameter then.

Using `sockaddr_in` might also provide the needed union to break apart the fields into the 4 octets. Probably wouldn't need to declare a union locally within the function.

Reply:
OK, I had assumed Winsock dependencies had intentially been left out of ErrorLog. I'm good with adding the dependency.

Reply:
Hmm, avoiding the dependence on socket code might have actually been part of the original design decision.

I'm not sure that was a good decision though.